monai/networks/nets/autoencoderkl.py

```
@@ -715,13 +715,25 @@ def load_old_state_dict(self, old_state_dict: dict, verbose=False) -> None:
+
+            if out_w in new_state_dict:
+                if proj_w in old_state_dict:
+                    new_state_dict[out_w] = old_state_dict.pop(proj_w)
+                    if proj_b in old_state_dict:
+                        new_state_dict[out_b] = old_state_dict.pop(proj_b)
+                else:
+                    # weights pre-date proj_attn: initialise to identity / zero
+                    new_state_dict[out_w] = torch.eye(new_state_dict[out_w].shape[0])
+                    new_state_dict[out_b] = torch.zeros(new_state_dict[out_b].shape)
+            elif proj_w in old_state_dict:
+                # new model has no out_proj at all – discard the legacy keys so they
```

Ruff (0.15.6)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)